### PR TITLE
Create AWS IAM

### PR DESCRIPTION
**Pull request by fritz-astronomer**

### CODE REVIEW

File: connections/prod/aws/iam [hunk 1]
```diff
@@ -0,0 +1,15 @@
+id: aws_iam
+connection_type: aws
+visible: true
+connection_name: AWS
+description: Configure a connection to AWS Services using an existing IAM Role.
+package: apache-airflow-providers-amazon
+parameters:
+  - airflow_param_name: role
```
Comment (melugoyal) [inline connections/prod/aws/iam:8]: this should be `role_arn`

**Review by jlaneve — COMMENTED** [review]
nice! do you mind starting w this in the dev folder? once it's merged + released I can take care of promoting through prod

File: connections/prod/aws/iam (hunk 1 above)
Comment (fritz-astronomer) [reply connections/prod/aws/iam:8]: appreciated 👍 (in my defense I was just trying to make this really fast and haven't taken the time to check anything 😅 )

File: connections/dev/aws/iam.yml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+id: aws_iam
+connection_type: aws
+visible: true
+connection_name: AWS
```
Comment (melugoyal) [inline connections/dev/aws/iam.yml:4]: this name should be distinct from the other aws connection, maybe call it "AWS Assume Role"?

Comment (fritz-astronomer) [reply connections/dev/aws/iam.yml:4]: This field is what shows up in the UI?

Comment (melugoyal) [reply connections/dev/aws/iam.yml]: sorry let me clarify -- there's both connection_name and method_name
image

the top text is connection_name, bottom text is method_name (meant to mean auth method). so i think the correct thing to do here would be connection_name=AWS, method_name="Assume Role" or similar

File: connections/dev/aws/iam.yml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+id: aws_iam
+connection_type: aws
+visible: true
+connection_name: AWS (Assume Role)
```
Comment (melugoyal) [inline connections/dev/aws/iam.yml:4]:
```suggestion
connection_name: AWS
method_name: Assume Role
```

**Review by melugoyal — APPROVED** [review]

File: connections/dev/aws/iam.yml [hunk 4]
```diff
@@ -0,0 +1,16 @@
+id: aws_iam
+connection_type: aws
+visible: true
+connection_name: AWS
+method_name: Assume Role
+description: Configure a connection to AWS Services using an existing IAM Role.
```
Comment (melugoyal) [inline connections/dev/aws/iam.yml:6]:
```suggestion
description: Configure a connection to AWS Services using an existing IAM Role. Ensure your Astro Deployment's Workload Identity has been authorized to assume this role via its trust relationship.
```

**Pull request merged**